src/js/components/projects/createForm.tsx

```diff
@@ -74,10 +88,130 @@ const ProjectForm = ({
+      const baseBranches = await apiFetch({
+        url: `${window.api_urls.repository_detail(
+          repository.id,
+        )}/feature_branches`,
+        dispatch,
+      });
+      setRepoBranches(baseBranches);
+      setFetchingBranches(false);
+      setBranchMenuOpen(true);
+    }
+  };
+
+  const handleBranchSelection = (selection: any) => {
+    setInputValue('');
+    setBaseBranch(selection[0].label);
+    setBranchMenuOpen(false);
+  };
+
+  const resetBranch = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    setInputValue('');
+    setBranchMenuOpen(false);
+    setFromBranchChecked(false);
+  };
+
+  const handleBranchChange = (value: string) => {
+    if (baseBranch) {
+      setBaseBranch(inputValue);
+    }
+    setInputValue(value);
+  };
+  const noOptionsFoundText = (
+    <p data-form="project-create">
+      {i18n.t("There aren't any available branches at this time.")}{' '}
+      <Button
+        label={i18n.t('Check Again')}
+        variant="base"
+        onClick={doGetBranches}
+      />
+      {i18n.t('to refresh this list or')}{' '}
+      <Button
+        label={i18n.t('start from a new branch')}
+        variant="link"
+        onClick={resetBranch}
+      />
+      .
+    </p>
+  );
+
+  const branchOptions = repoBranches.map((item, index) => ({
+    id: `${index + 1}`,
+    label: item,
+  }));
+
   return (
     <form onSubmit={handleSubmit} className="slds-form slds-m-bottom--large">
       {isOpen && (
         <>
+          <Checkbox
+            id="project-base-branch"
+            assistiveText={{
+              label: `${i18n.t('Use existing Github branch')}`,
+            }}
+            labels={{
+              label: `${i18n.t('Use existing Github branch')}`,
+            }}
+            checked={fromBranchChecked}
+            className="slds-form-element_stacked slds-p-left_none"
+            onChange={(
+              event: React.ChangeEvent<HTMLInputElement>,
+              { checked }: { checked: boolean },
+            ) => handleChange(checked)}
+          />
+          {fromBranchChecked && (
+            <Combobox
+              id="combobox-inline-single"
+              isOpen={branchMenuOpen}
+              events={{
+                onRequestOpen: doGetBranches,
+                onSelect: (event: React.MouseEvent, data: any) =>
+                  handleBranchSelection(data.selection),
+                onChange: (
+                  event: React.FormEvent<HTMLInputElement>,
+                  {
+                    value,
+                  }: {
+                    value: string;
+                  },
+                ) => handleBranchChange(value),
+              }}
+              labels={{
+                label: `${i18n.t('Select a branch to use for this project')}`,
+                noOptionsFound: noOptionsFoundText,
+              }}
+              menuItemVisibleLength={5}
+              options={comboboxFilter({
```

Comment:
@jgerigmeyer getting an error from TS about using this filter from slds if you can take a peek. A workaround for it is not obvious to me.  
![Screenshot from 2020-05-06 17-05-01](https://user-images.githubusercontent.com/12376362/81228772-ad292580-8fbc-11ea-83be-cbd7fcbfea47.png)

Reply:
@duggiemitchell Resolved in https://github.com/oddbird/Metecho/pull/268/commits/62a4fb1bb73d48f589931c08a31156ee30d39df6

It looks like `selection` is expected to be an array of item objects (similar to `options`), so there's still a warning about that.